### PR TITLE
Tooltips for geom_text, line pattern in label_fomat

### PR DESCRIPTION
**Pull request by OLarionova-HORIS**

### CODE REVIEW

File: plot-base-portable/src/commonMain/kotlin/jetbrains/datalore/plot/base/util/StringFormat.kt [hunk 1]
```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2020. JetBrains s.r.o.
+ * Use of this source code is governed by the MIT license that can be found in the LICENSE file.
+ */
+
+package jetbrains.datalore.plot.base.util
```
Comment (alshan) [inline plot-base-portable/src/commonMain/kotlin/jetbrains/datalore/plot/base/util/StringFormat.kt:6]: We already have some formatting-related code in 'base-portable' (NumberFormat) and in 'plot-common-portable' (text).

If StringFormat is not too plot-dependent - consider moving to ... 'base-portable' maybe?

File: plot-base-portable/src/jvmTest/kotlin/plot/base/StringFormatTest.kt [hunk 2]
```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2020. JetBrains s.r.o.
+ * Use of this source code is governed by the MIT license that can be found in the LICENSE file.
+ */
+
+package jetbrains.datalore.plot.base
```
Comment (alshan) [inline plot-base-portable/src/jvmTest/kotlin/plot/base/StringFormatTest.kt:6]: Should be in ...base.util package.

**Review by alshan — CHANGES_REQUESTED** [review]

File: plot-base-portable/src/commonMain/kotlin/jetbrains/datalore/plot/base/util/StringFormat.kt (hunk 1 above)
Comment (OLarionova-HORIS) [reply plot-base-portable/src/commonMain/kotlin/jetbrains/datalore/plot/base/util/StringFormat.kt]: StringFormat class was moved to 'base-portable'.

File: plot-base-portable/src/jvmTest/kotlin/plot/base/StringFormatTest.kt (hunk 2 above)
Comment (OLarionova-HORIS) [reply plot-base-portable/src/jvmTest/kotlin/plot/base/StringFormatTest.kt]: StringFormatTest is also moved: _base-portable/src/**jvmTest**/kotlin/base/stringFormat/StringFormatTest.kt_. Is it ok?

Comment (alshan) [reply plot-base-portable/src/jvmTest/kotlin/plot/base/StringFormatTest.kt]: Yes, that's fine.

**Pull request merged**